src/nvim/terminal.c

```diff
@@ -1234,7 +1234,8 @@ static void refresh_screen(Terminal *term, buf_T *buf)
 
   int change_start = row_to_linenr(term, term->invalid_start);
   int change_end = change_start + changed;
-  changed_lines(change_start, 0, change_end, added);
+  // Note: don't send nvim_buf_lines_event event for a :terminal buffer
+  changed_lines(change_start, 0, change_end, added, false);
```

Comment:
@KillTheMule is there at test to check that `:terminal` buffers do not trigger `nvim_buf_lines_event`?

Reply:
No there isn't. But now that you mention it: Why shouldn't they? I mean, they're buffers after all, if someone wants to subscribe to their updates, I don't see I reason why we should deny them. Or is there a specific reason I'm not seeing?

Reply:
Sure, why not. I was just going by the comment here.

Would still want a test either way

Reply:
I'll send a PR, might take some time though.

Reply:
I want to monitor the content of a terminal buffer that loaded from session file,  it seems no other options except using the buffer updates API.